### ISSUE TITLE
If the pod is in CrashLoopBackOff, the phase is reported as Running

### ISSUE THREAD

**Issue by lburgazzoli** [apache/camel-k#2445]
I have the following scenarion:

```
➜ k tree klb cd208ccf823b044089c48669c1ec1d6e4
NAMESPACE  NAME                                                                    READY  REASON              AGE 
default    KameletBinding/cd208ccf823b044089c48669c1ec1d6e4                        True                       143m
default    └─Integration/cd208ccf823b044089c48669c1ec1d6e4                         False  ReplicaSetNotReady  143m
default      ├─ConfigMap/cd208ccf823b044089c48669c1ec1d6e4-application-properties  -                          143m
default      ├─ConfigMap/cd208ccf823b044089c48669c1ec1d6e4-source-000              -                          143m
default      ├─Deployment/cd208ccf823b044089c48669c1ec1d6e4                        -                          143m
default      │ └─ReplicaSet/cd208ccf823b044089c48669c1ec1d6e4-585565bdf5           -                          143m
default      │   └─Pod/cd208ccf823b044089c48669c1ec1d6e4-585565bdf5-nptc7          False  ContainersNotReady  143m
default      └─IntegrationKit/kit-cd208ccf823b044089c48669c1ec1d6e4                -                          143m
```

so the POD is not ready for an internal error but the phase of the integration is still reported as `Running` and the kamelet binding as `Ready`:

```
➜ k get klb cd208ccf823b044089c48669c1ec1d6e4
NAME                                PHASE
cd208ccf823b044089c48669c1ec1d6e4   Ready
➜ k get it cd208ccf823b044089c48669c1ec1d6e4
NAME                                PHASE     KIT   REPLICAS
cd208ccf823b044089c48669c1ec1d6e4   Running         1
➜ k get pods | grep cd208ccf823b044089c48669c1ec1d6e4
cd208ccf823b044089c48669c1ec1d6e4-585565bdf5-nptc7   0/1     CrashLoopBackOff   32         141m
```

I wonder if we can ave a better way to report that the pod is not running.
By looking at the integration resource only, one may mistakenly think the integration is running but it is not.

**Comment by lburgazzoli** [comment]
Additional info:

The _Integration_ has a condition where type=ready and status=false

```yaml
apiVersion: camel.apache.org/v1
kind: Integration
metadata:
  name: cd208ccf823b044089c48669c1ec1d6e4
status:
  conditions:
  - lastTransitionTime: "2021-06-24T20:49:04Z"
    lastUpdateTime: "2021-06-24T20:49:04Z"
    reason: ReplicaSetNotReady
    status: "False"
    type: Ready
  phase: Running
```

But the parent _KameletBinding_, does not reflect such status:

````yaml
apiVersion: camel.apache.org/v1alpha1
kind: KameletBinding
metadata:
  name: cd208ccf823b044089c48669c1ec1d6e4
spec:
status:
  conditions:
  - lastTransitionTime: "2021-06-24T18:07:56Z"
    lastUpdateTime: "2021-06-24T18:07:56Z"
    status: "True"
    type: Ready
  phase: Ready
```

**Comment by claudio4j** [comment]
Just yesterday [I pushed a change](https://github.com/apache/camel-k/pull/2292) to update integration status related to Deployment status, which in this case the kameletbinding and integration sets the Phase as Error.
```
$ k get it,klb,pods 
NAME                                                  PHASE   KIT                        REPLICAS
integration.camel.apache.org/mongodb-source-binding   Error   kit-c3ahd39gvo520gho5ge0   1

NAME                                                     PHASE
kameletbinding.camel.apache.org/mongodb-source-binding   Error

NAME                                          READY   STATUS             RESTARTS   AGE
pod/camel-k-operator-5f55d8b4c5-95kqk         1/1     Running            2          23h
pod/mongodb-server-7f65f49f79-vq2k2           1/1     Running            12         8d
pod/mongodb-source-binding-5647f6c7dd-77kd9   0/1     CrashLoopBackOff   6          8m58s
```
Can you try with a recent camel-k-operator build ?

**Comment by astefanutti** [comment]
It should be fixed with #2682, available in 1.7.0.
